### ISSUE TITLE
Video freezes, audio keeps playing

### ISSUE THREAD

**Issue by jimbo1111** [moonlight-stream/moonlight-qt#754]
I've been getting random video crashes for a while now, Moonlight will work just fine but at random times the video stream will freeze but audio will continue, and the only fix is to quit the stream and start it up again.  After restarting stream, graphics will perform poorly - whereas I was at a solid 60fps before, the game will now run between 40-50fps (according to RTSS - so the frame drops are happening on the host, not on the client) despite GPU being at low utilization percentage.

Host is a 3080 headless pc on win10 with 512.15 and GeForce Experience 3.25.0.84 and windows default antivirus, client is an M1 Max MacBook Pro on MacOS 12.3 and all have Moonlight 3.2.0. 

This has happened with Civ 6, Elden Ring, Forza Horizon 5, and Dicey Dungeons - the last of these is an OpenGL game, so it's not only DX12 games doing this.  Most of this testing was on 511.79, but just tested Elden Ring on 512.15 and it still exhibits the same issue.  In fact, just after installing the driver update, I played Ghostwire for a few minutes (without rebooting) and the computer crashed to reboot.  Then tried Elden Ring and it showed the issue as stated above.

edit: it just happened again, but this time, instead of crashing to reboot, it crashed to bluescreen (which I found out by being unable to restart the stream, so I plugged in a monitor and saw "your device ran into a problem and needs to restart" and 0% progress on collecting error info.  error code is CLOCK_WATCHDOG_TIMEOUT.  I have not witnessed this before, but I also have not had a monitor that plugs into this computer until this week).

It can be reproduced by playing any game for several minutes, after which the issue will happen.  It does not seem to be associated with any sort of in-game activity - I can be in a menu, or in a graphically intense scene, etc.

None of this happens in Parsec.  I would normally just use that instead of Moonlight, but I am also having encoding issues with it, with high encoding times (>1 frame) causing stutters.  Moonlight does not have these stutters, but does crash after a few minutes of play.  I would like at least one of these programs to work properly.

Settings (top is client, bottom is host): https://imgur.com/a/QhPql8Q

I have asked multiple times in the Discord and never received any response.

Github is not letting me post logs (says the file is empty), so here's some pastes. I'm not sure if these are from the same session or not, it doesn't seem to have made a windows log for the last session, according to file creation times...

host log: https://pastebin.com/WXZ4HQHr

client log: https://pastebin.com/jEp5u79a

**Comment by cgutman** [comment]
> I've been getting random video crashes for a while now, Moonlight will work just fine but at random times the video stream will freeze but audio will continue, and the only fix is to quit the stream and start it up again.

Have you tried disabling hardware-accelerated GPU scheduling on the host? https://github.com/moonlight-stream/moonlight-docs/wiki/Troubleshooting#if-you-have-issues-with-the-video-getting-stuck-while-streaming-disable-hardware-accelerated-gpu-scheduling-on-your-host-pc

> host log: https://pastebin.com/WXZ4HQHr

This looks like a log of Moonlight crashing? (Are you running Moonlight on your host by mistake?)

I need the logs from GeForce Experience (C:\ProgramData\NVIDIA Corporation\NvStream) from the host side. 

> client log: https://pastebin.com/jEp5u79a

I don't see an indication of an unexpected disconnection in this log. It looks like a graceful disconnect initiated after you pressed the quit key combo ("Detected quit key combo"). It was also acknowledged by the host ("ENet peer acknowledged disconnection") which is atypical of an unexpected disconnection (which would have been initiated by the host and generate a different log message).

```
00:17:36 - SDL Info (0): Detected quit key combo
00:17:36 - SDL Info (0): Quit event received
00:17:36 - SDL Info (0): Raising 3 keys
00:17:36 - SDL Info (0): Global video stats
00:17:36 - SDL Info (0): ----------------------------------------------------------
Incoming frame rate from network: 59.65 FPS
Decoding frame rate: 59.65 FPS
Rendering frame rate: 59.31 FPS
Frames dropped by your network connection: 0.06%
Frames dropped due to network jitter: 0.56%
Average network latency: 2 ms (variance: 1 ms)
Average decoding time: 4.99 ms
Average frame queue delay: 3.88 ms
Average rendering time (including monitor V-sync latency): 14.68 ms
00:17:36 - SDL Info (0): Stopping input stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Stopping audio stream...
00:17:36 - SDL Info (0): ENet wait interrupted
00:17:36 - SDL Info (0): Control stream connection failed: -1
00:17:36 - Qt Info: Found "gamecontrollerdb.txt" at "/Users/jimbo1111/Library/Caches/Moonlight Game Streaming Project/Moonlight/gamecontrollerdb.txt"
00:17:36 - SDL Info (0): Loaded 64 new gamepad mappings
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Stopping video stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Stopping control stream...
00:17:36 - SDL Info (0): ENet peer acknowledged disconnection
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Cleaning up input stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Cleaning up video stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Cleaning up control stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Cleaning up audio stream...
00:17:36 - SDL Info (0): done
00:17:36 - SDL Info (0): Cleaning up platform...
00:17:36 - SDL Info (0): done
```

> edit: it just happened again, but this time, instead of crashing to reboot, it crashed to bluescreen (which I found out by being unable to restart the stream, so I plugged in a monitor and saw "your device ran into a problem and needs to restart" and 0% progress on collecting error info. error code is CLOCK_WATCHDOG_TIMEOUT. I have not witnessed this before, but I also have not had a monitor that plugs into this computer until this week).

`CLOCK_WATCHDOG_TIMEOUT` indicates you've got some other problem going on. That's going to most likely be a hardware, firmware/BIOS, or driver issue. I would suggest going back to stock frequency and voltage if you have any CPU or GPU overclocks, updating to the latest BIOS, etc. Streaming from the PC using Moonlight might trigger the BSOD, but it's certainly not Moonlight's fault. It's just exposing instability that already exists (perhaps with NvFBC, which GeForce Experiences uses for game capture).

**Comment by jimbo1111** [comment]
Right, I used the quit keys after the video froze and it was still streaming audio.  So it would look like a graceful exit, but I had to gracefully exit because the video was frozen.

Just now I reset my GPU underclock with MSI Afterburner and turned off GPU scheduling (didn't change any BIOS though), then tried playing Ghostwire.  Tried to do a particular move, and it crashed to reboot (nvstreamer log below).  Tried playing Elden Ring for a bit, and did not experience the video freezing issue, then went back to Ghostwire and did the same move in the same place, and it crashed to reboot again.  Then tried doing the same through Parsec instead of Moonlight, and the computer did not crash to reboot, it just kept playing the game normally.

Here is the paste of NvStreamerCurrent.log right after the first crash to reboot: https://pastebin.com/eS7shCvk

If I experience another video freeze I'll post the nvstreamer log, but perhaps the gpu scheduling will fix that issue.  But I guess this is about another issue then, the crash to reboot that seems to be happening in Moonlight and not other headless streaming apps.
